editors/code/src/util.ts

```diff
@@ -14,21 +16,46 @@ export function assert(condition: boolean, explanation: string): asserts conditi
 
 export const log = new class {
     private enabled = true;
+    private readonly output = vscode.window.createOutputChannel("Rust Analyzer Client");
```

Comment:
This feels like it should be disposable? I think global singleton logger is the right pattern, so let’s dispose manually on deactivate

Reply:
We cannot do this manually in `deactivate()`. We call `activate()` and `deactivate()` manually in `Restart server` command which means we will dispose the singleton and try to use it further. I just added it to `subscriptions`: 13872543e074adc153b440660beda441fd562f53

Reply:
Or wait, this is a mistake too...

Reply:
We dispose subscriptions in `Restart server` too. So we need to explicitly initialize the logger

Reply:
But then do we really want to wipe out the logs in `Restart server`? I don't think so, maybe we can leave this leak? Btw, we never dispose debug output channel too:
https://github.com/rust-analyzer/rust-analyzer/blob/master/editors/code/src/debug.ts#L10